core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/project/ProjectResource.scala

```diff
@@ -242,7 +252,9 @@ class ProjectResource {
       @PathParam("pid") pid: UInteger,
       @PathParam("wid") wid: UInteger
   ): Unit = {
-    workflowOfProjectDao.insert(new WorkflowOfProject(wid, pid))
+    if (!workflowOfProjectExists(wid, pid)) {
+      workflowOfProjectDao.insert(new WorkflowOfProject(wid, pid))
+    }
```

Comment:
Should we throw an error for the "else" case?

Reply:
I originally didn't have it throw an error since in [navigation.ts](https://github.com/Texera/texera/pull/1461/files#diff-a2405c8da3a8489eb535b4147bb55dbf78746dac75b9e188ad2c2d0953eab5a0), creating / persisting workflow are both done (#1458) by the persistWorkflow() function and call the same endpoint to add workflows to project.  If the backend threw an error in the "else" case, the frontend would need to separate creating / persisting a workflow or check if the workflow is already in the project before adding it.  I remember you mentioned checking before adding may risk the atomicity of adding workflows to projects.  What do you recommend for approaching throwing an error in this case?

Reply:
got it. From my understanding, if a workflow already exists in the target project, this function will do nothing for now. I think it's ok to not throw an error.

Reply:
Ok, will mark as resolved for now, thank you